### PR TITLE
Fix : some change in mvp architecture of forgotpassword activity and refactoring

### PR DESCRIPTION
**Pull request by amitiwary999**
Fixes issue #826

### CODE REVIEW

File: app/src/main/java/org/fossasia/susi/ai/data/contract/IForgotPasswordModel.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * Created by meeera on 10/7/17.
+ */
+interface IForgotPasswordModel {
+
+    interface onFinishListener {
+        fun onError(throwable: Throwable)
+        fun onSuccess(response: Response<ForgotPasswordResponse>)
+    }
+
+    fun signup(email: String, listener: IForgotPasswordModel.onFinishListener)
```
Comment (mayank408) [inline app/src/main/java/org/fossasia/susi/ai/data/contract/IForgotPasswordModel.kt]: Pls change the function names

File: app/src/main/java/org/fossasia/susi/ai/data/contract/IForgotPasswordModel.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    fun cancelSignup()
```
Comment (mayank408) [inline app/src/main/java/org/fossasia/susi/ai/data/contract/IForgotPasswordModel.kt]: here too

**Comment by amitiwary999** [comment]
@chiragw15 @mayank408 please review

**Comment by chiragw15** [comment]
@amitiwary999 Please pull the latest code and make changes accordingly. There are some methods which have already been implemented now, so you can omit them.

**Comment by amitiwary999** [comment]
@chiragw15 @mayank408 review

File: app/src/main/java/org/fossasia/susi/ai/forgotPassword/contract/IForgotPasswordPresenter.kt [hunk 3]
```diff
@@ -1,14 +1,15 @@
-package org.fossasia.susi.ai.forgotPassword
+package org.fossasia.susi.ai.data.contract
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/forgotPassword/contract/IForgotPasswordPresenter.kt]: This is wrong package name.

File: app/src/main/java/org/fossasia/susi/ai/forgotPassword/contract/IForgotPasswordPresenter.kt [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
 
 import android.content.Context
+import org.fossasia.susi.ai.data.contract.forgotPassword.IForgotPasswordView
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/forgotPassword/contract/IForgotPasswordPresenter.kt]: View should not be in data package

File: app/src/main/java/org/fossasia/susi/ai/forgotPassword/contract/IForgotPasswordPresenter.kt [hunk 5]
```diff
@@ -1,14 +1,15 @@
-package org.fossasia.susi.ai.forgotPassword
+package org.fossasia.susi.ai.data.contract
 
 import android.content.Context
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/forgotPassword/contract/IForgotPasswordPresenter.kt]: unused

File: app/src/main/java/org/fossasia/susi/ai/forgotPassword/contract/IForgotPasswordView.kt [hunk 6]
```diff
@@ -0,0 +1,12 @@
+package org.fossasia.susi.ai.data.contract.forgotPassword
```
Comment (chiragw15) [inline app/src/main/java/org/fossasia/susi/ai/forgotPassword/contract/IForgotPasswordView.kt]: wrong package name

**Pull request merged**